juju1/provider/openstack/provider.go

```diff
@@ -1703,3 +1704,43 @@ func (e *environ) TagInstance(id instance.Id, tags map[string]string) error {
 	}
 	return nil
 }
+
+func (e *environ) changeTags(tags map[string]string) error {
+	instances, err := e.AllInstances()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	for _, inst := range instances {
+		err = e.TagInstance(inst.Id(), tags)
+		// Not sure - maybe we should attempt for all instances.
```

Comment:
seems fine to me, I think it has to be all or nothing

Reply:
Cool, thanks.